cli/bench/bench.go

```diff
@@ -0,0 +1,118 @@
+package bench
+
+import (
+	"fmt"
+	"math/rand"
+	"sync"
+	"time"
+
+	"github.com/FZambia/tarantool"
+	"github.com/tarantool/cartridge-cli/cli/context"
+)
+
+func PrintResults(results Results) {
+	fmt.Printf("\nResults:\n")
+	fmt.Printf("\tSuccess operations: %d\n", results.successResult)
+	fmt.Printf("\tFailed  operations: %d\n", results.failedResult)
+	fmt.Printf("\tRequest count: %d\n", results.hanledRequestsCount)
+	fmt.Printf("\tTime (seconds): %f\n", results.duration)
+	fmt.Printf("\tRequests per second: %d\n\n", results.requestsPerSecond)
+}
+
+func SpacePreset(ctx context.BenchCtx, tarantoolConnection *tarantool.Connection) error {
+	DropSpace(ctx, tarantoolConnection)
+	if err := CreateSpace(tarantoolConnection); err != nil {
+		return err
+	}
+	return nil
+}
+
+func CountRequest(err error, results *Results) {
+	if err == nil {
+		results.successResult++
+	} else {
+		results.failedResult++
+	}
+	results.hanledRequestsCount++
+}
+
+func ConnectionLoop(ctx context.BenchCtx, requestsLoop requestLoopType, results *Results, waitGroup *sync.WaitGroup, benchActiveFlag *bool) {
+	tarantoolConnection, err := TarantoolConnect(ctx)
+	if err != nil {
+		return
+	}
+	defer tarantoolConnection.Close()
+
+	var connectionWait sync.WaitGroup
+	for requestId := 0; requestId < ctx.SimultaneousRequests; requestId++ {
+		connectionWait.Add(1)
+		go requestsLoop(ctx, tarantoolConnection, results, &connectionWait, benchActiveFlag)
+	}
+
+	connectionWait.Wait()
+	defer waitGroup.Done()
+}
+
+func Run(ctx context.BenchCtx) error {
+	rand.Seed(time.Now().UnixNano())
+
+	timerDuration := ctx.Duration
+
+	operationsMap := []Operation{
+		{"insert", ctx.InsertCount, InsertRequestsLoop, 0, 0},
+		{"delete", ctx.DeleteCount, DeleteRequestsLoop, 0, 0},
+		{"update", ctx.UpdateCount, UpdateRequestsLoop, 0, 0},
+		{"select", ctx.SelectCount, SelectRequestsLoop, 0, 0},
+	}
+
+	allOperationCount := 0
+	for _, operation := range operationsMap {
+		allOperationCount += operation.count
+	}
+
+	maxConnectionsCount := ctx.Connections
+	for i := 0; i < len(operationsMap); i++ {
+		operationsMap[i].percent = float64(operationsMap[i].count) / float64(allOperationCount)
+		operationsMap[i].connectionsCount = int(float64(maxConnectionsCount) * operationsMap[i].percent)
+		if operationsMap[i].count != 0 && operationsMap[i].connectionsCount == 0 {
+			operationsMap[i].connectionsCount = 1
+		}
+	}
+
+	tarantoolConnection, err := TarantoolConnect(ctx)
+	if err != nil {
+		return err
+	}
+	PrintConfig(ctx, tarantoolConnection, operationsMap)
+
+	var waitGroup sync.WaitGroup
+	results := Results{}
+
+	if err := SpacePreset(ctx, tarantoolConnection); err != nil {
+		return err
+	}
+	fmt.Printf("Benchmark start\n")
+	fmt.Printf("...\n")
```

Comment:
fmt.Println. In general, we use log.Infof

Reply:
We will think about output format later

Reply:
What do you mean "later"? In the project, ``log.Infof`` is used everywhere for informational messages (like this one). I don't understand why you can't add it here.